nautobot_golden_config/tables.py

```diff
@@ -227,27 +201,71 @@ class GoldenConfigTable(BaseTable):
     """Table to display Config Management Status."""
 
     pk = ToggleColumn()
-    device__name = TemplateColumn(
-        template_code="""<a href="{% url 'dcim:device' pk=record.device.pk %}">{{ record.device }}</a>""",
+    name = TemplateColumn(
+        template_code="""<a href="{% url 'dcim:device' pk=record.pk %}">{{ record.name }}</a>""",
         verbose_name="Device",
     )
+
     if ENABLE_BACKUP:
-        backup_last_success_date = TemplateColumn(verbose_name="Backup Status", template_code=BACKUP_SUCCESS)
+        backup_last_success_date = Column(verbose_name="Backup Status", empty_values=())
     if ENABLE_INTENDED:
-        intended_last_success_date = TemplateColumn(verbose_name="Intended Status", template_code=INTENDED_SUCCESS)
+        intended_last_success_date = Column(verbose_name="Intended Status", empty_values=())
     if ENABLE_COMPLIANCE:
-        compliance_last_success_date = TemplateColumn(
-            verbose_name="Compliance Status", template_code=COMPLIANCE_SUCCESS
-        )
+        compliance_last_success_date = Column(verbose_name="Compliance Status", empty_values=())
 
     actions = TemplateColumn(
         template_code=ALL_ACTIONS, verbose_name="Actions", extra_context=CONFIG_FEATURES, orderable=False
     )
 
+    def render_backup_last_success_date(self, record, column):  # pylint: disable=no-self-use
```

Comment:
What about roughly 

```
    def _render_last_success_date(self, record, column, value):  # pylint: disable=no-self-use
        """Abstract method to get last success per row record."""
        entry = record.goldenconfig_set.first()
        last_success_date = getattr(entry, f"{value}_last_success_date", None)
        last_attempt_date = getattr(entry, f"{value}_last_attempt_date", None)
        if not last_success_date or not last_attempt_date:
            column.attrs = {"td": {"style": "color:black"}}
            return "--"
        if not last_success_date and not last_attempt_date:
            column.attrs = {"td": {"style": "color:black"}}
            return "--"
        if last_success_date and last_attempt_date == last_success_date:
            column.attrs = {"td": {"style": "color:green"}}
            return last_success_date
        column.attrs = {"td": {"style": "color:red"}}
        return last_success_date


    def render_backup_last_success_date(self, record, column):  # pylint: disable=no-self-use
        """Pull back backup last success per row record."""
        return _render_last_success_date(self, record, column, "backup")

    def render_intended_last_success_date(self, record, column):  # pylint: disable=no-self-use
        """Pull back intended last success per row record."""
        return _render_last_success_date(self, record, column, "intended")

    def render_compliance_last_success_date(self, record, column):  # pylint: disable=no-self-use
        """Pull back compliance last success per row record."""
        return _render_last_success_date(self, record, column, "compliance")
```

Reply:
Thing of beauty @itdependsnetworks , thanks.  I'll update the return lines to do self._rend... and remove self from the function call.  But I'll implement this today.